09_object_scope.rb

```diff
@@ -14,10 +14,13 @@ def name
 
   def greet(other_person)
     #This local variable shadows the method 'name'
-    name = "Not my name"
+    name = @name
     # TODO: We have a name clash betweeen a local variable and a method name
     # TODO: How can we be specific?
+    # => Indem innerhalb der Methode greet nicht der Wert "name" verwendet wird,
+    # da dieser Wert außerhalb definiert wurde und somit übernommen wird.
     # TODO :How can we call a method on the current object?
+    # => Umbenennen, da es sonst zu einer Nameskollision kommt?!
```

Comment:
There is a way! You can use the keyword `self`. Check [this](http://ruby-for-beginners.rubymonstas.org/writing_classes/self.html) out.